src/Files.App/ServicesImplementation/Settings/FoldersSettingsService.cs

```diff
@@ -137,6 +187,36 @@ public bool ShowSizeColumn
 			set => Set(value);
 		}
 
+		public bool ShowGitStatusColumn
+		{
+			get => Get(true);
+			set => Set(value);
+		}
+
+		public bool ShowGitLastCommitDateColumn
+		{
+			get => Get(true);
+			set => Set(value);
+		}
+
+		public bool ShowGitLastCommitMessageColumn
+		{
+			get => Get(true);
+			set => Set(value);
+		}
+
+		public bool ShowGitCommitAuthorColumn
+		{
+			get => Get(true);
+			set => Set(value);
+		}
+
+		public bool ShowGitLastCommitShaColumn
+		{
+			get => Get(true);
+			set => Set(value);
+		}
```

Comment:
Maybe these should be set to false by default (at least until we have behavior to auto hide the non git columns).

Reply:
We already have it. If the dir is not git repo, git columns will be hidden. And if it is git repo, they will be shown.

Reply:
The non git columns are still visible, until we have behavior to auto hide the other columns, I think we should limit which columns are being added to save space.

Reply:
Let's forget this for now.